chall05/gbudau.c

```diff
@@ -0,0 +1,29 @@
+#include <ctype.h>
+
+int	ft_ie_except_after_c(char *str)
+{
+	char *start;
+
+	start = str;
+	while (*str)
+	{
+		if (tolower(*str) == 'c')
+		{
+			if (*(str + 1) && tolower(*(str + 1)) == 'i' &&
+							tolower(*(str + 2)) == 'e')
+				return (0);
+		}
+		else if (tolower(*str) == 'e')
+		{
+			if (str - start > 0 && tolower(*(str - 1)) == 'c')
+			{
+				str++;
+				continue;
+			}
+			if (tolower(*(str + 1)) == 'i')
+				return (0);
```

Comment:
Having to use `ft_lower` function prevents you from doing so, but you could have used `strstr` to look for the wanted occurrences. For example you know that you should never find `cie`, this could be solved in an `if`.  
Later you would still have to loop to check for `ei` that are not preceded with `c`.

You also could have organized your conditions differently, I must say that this is really hard to read. Maybe putting some of the checks in an additional fonction would have eased the reading too.

Reply:
Sorry about the readability, i made another version based on your sugestions:

```
#include <bsd/string.h>

/*
** Testing it on Linux so im including BSD library
** and linking with -lbsd when compiling for strnstr
** 
** On MacOS should be '#include <string.h>'
*/

int	ft_ie_except_after_c(char *str)
{
	size_t	i;

	i = 0;
	while (str[i])
	{
		if (strnstr(&str[i], "cie", 3))
				return (0);
		else if (str[i] == 'e')
		{
			if (i && str[i - 1] == 'c')
			{
				i++;
				continue;
			}
			if (str[i + 1] == 'i')
				return (0);
		}
		i++;
	}
	return (1);
}
```

Also thank you for the feedback :heart: